packages/lib/src/artifacts/ZosContract.ts

```diff
@@ -0,0 +1,111 @@
+import BN from 'bignumber.js';
+import ZWeb3 from './ZWeb3';
+import { getSolidityLibNames, hasUnlinkedVariables } from '../utils/Bytecode';
+import { Contract, TransactionObject, BlockType } from 'web3-eth-contract';
+import { Callback, EventLog, EventEmitter, TransactionReceipt } from 'web3/types';
+import Contracts from './Contracts';
+import { StorageLayoutInfo } from '../validations/Storage';
+import _ from 'lodash';
+
+/*
+ * ZosContract is an interface that extends Web3's Contract interface, adding some properties and methods like:
+ * address getter: retrieves the deployed address
+ * schema: compilation artifacts
+ * new(): deployes a new contract
+ * at(): retrieves a deployed contract at the specified address
+ * link(): links libraries in a contract schema
+ */
+export default interface ZosContract {
+
+  // Web3 Contract interface.
```

Comment:
Is it possible to load these typings from web3 and having ZosContract extend from that interface, to avoid copy-pasting?

Reply:
Unfortunately, Web3's existing declarations define a class instead of an interface for the Contract type, and Typescript doesn't allow to extend clases from `.d.ts` files. Believe me, I tried everything! We could raise another issue in Web3 with this request.

Reply:
Damn! Yes, let's open an issue then, though I wouldn't expect any answer soon; I think they have more pressing ones.